geode-core/src/main/java/org/apache/geode/internal/cache/persistence/PersistenceAdvisorImpl.java

```diff
@@ -510,18 +510,18 @@ public PersistentMembershipView getMembershipView() {
   public boolean checkMyStateOnMembers(Set<InternalDistributedMember> replicates)
       throws ReplyException {
     PersistentStateQueryResults remoteStates = getMyStateOnMembers(replicates);
+    Set<InternalDistributedMember> copyOfReplicates = new HashSet<>(replicates);
```

Comment:
Instead of always having this code create a copy every time it is called, start the copyofReplicates out as null. Then, in the exceptional case when we need to remove from it (line 540), you can ask if it is null and then make it a copy of  replicates.